### PR TITLE
Extend the benchmarks to DSIGN

### PR DESCRIPTION
**Pull request by dcoutts**
And tidy up the use of the benchmark data. Eliminate the duplication.

### CODE REVIEW

**Comment by dcoutts** [comment]
Rebased on current master. I think this is ready to go.

Results on a AMD Ryzen 7 3700X:
```
benchmarking DSIGN/Ed25519/genKeyDSIGN
time                 17.87 μs   (17.84 μs .. 17.90 μs)
                     1.000 R²   (1.000 R² .. 1.000 R²)
mean                 17.88 μs   (17.86 μs .. 17.89 μs)
std dev              48.32 ns   (39.84 ns .. 61.47 ns)

benchmarking DSIGN/Ed25519/signDSIGN
time                 18.77 μs   (18.76 μs .. 18.79 μs)
                     1.000 R²   (1.000 R² .. 1.000 R²)
mean                 18.82 μs   (18.79 μs .. 18.88 μs)
std dev              130.4 ns   (88.07 ns .. 221.3 ns)

benchmarking DSIGN/Ed25519/verifyDSIGN
time                 49.47 μs   (49.42 μs .. 49.54 μs)
                     1.000 R²   (1.000 R² .. 1.000 R²)
mean                 49.57 μs   (49.49 μs .. 49.80 μs)
std dev              433.3 ns   (211.1 ns .. 882.3 ns)

benchmarking DSIGN/EcdsaSecp256k1/genKeyDSIGN
time                 94.09 ns   (93.95 ns .. 94.20 ns)
                     1.000 R²   (1.000 R² .. 1.000 R²)
mean                 94.08 ns   (93.85 ns .. 94.25 ns)
std dev              677.8 ps   (442.6 ps .. 981.2 ps)

benchmarking DSIGN/EcdsaSecp256k1/signDSIGN
time                 27.91 μs   (27.85 μs .. 27.96 μs)
                     1.000 R²   (1.000 R² .. 1.000 R²)
mean                 27.84 μs   (27.81 μs .. 27.87 μs)
std dev              102.8 ns   (85.41 ns .. 135.9 ns)

benchmarking DSIGN/EcdsaSecp256k1/verifyDSIGN
time                 35.28 μs   (35.27 μs .. 35.29 μs)
                     1.000 R²   (1.000 R² .. 1.000 R²)
mean                 35.30 μs   (35.28 μs .. 35.32 μs)
std dev              74.92 ns   (60.45 ns .. 96.41 ns)

benchmarking DSIGN/SchnorrSecp256k1/genKeyDSIGN
time                 36.30 ns   (36.06 ns .. 36.58 ns)
                     1.000 R²   (1.000 R² .. 1.000 R²)
mean                 36.46 ns   (36.29 ns .. 36.65 ns)
std dev              581.9 ps   (496.1 ps .. 669.6 ps)
variance introduced by outliers: 20% (moderately inflated)

benchmarking DSIGN/SchnorrSecp256k1/signDSIGN
time                 44.00 μs   (43.92 μs .. 44.15 μs)
                     1.000 R²   (1.000 R² .. 1.000 R²)
mean                 44.04 μs   (43.97 μs .. 44.14 μs)
std dev              275.8 ns   (218.2 ns .. 398.8 ns)

benchmarking DSIGN/SchnorrSecp256k1/verifyDSIGN
time                 36.40 μs   (36.34 μs .. 36.47 μs)
                     1.000 R²   (1.000 R² .. 1.000 R²)
mean                 36.41 μs   (36.37 μs .. 36.48 μs)
std dev              173.3 ns   (124.6 ns .. 281.0 ns)

benchmarking KES/Sum6KES/genKey
time                 1.202 ms   (1.199 ms .. 1.207 ms)
                     1.000 R²   (1.000 R² .. 1.000 R²)
mean                 1.208 ms   (1.206 ms .. 1.211 ms)
std dev              9.403 μs   (7.121 μs .. 11.97 μs)

benchmarking KES/Sum6KES/signKES
time                 18.84 μs   (18.80 μs .. 18.88 μs)
                     1.000 R²   (1.000 R² .. 1.000 R²)
mean                 18.81 μs   (18.78 μs .. 18.85 μs)
std dev              115.2 ns   (87.00 ns .. 152.9 ns)

benchmarking KES/Sum6KES/verifyKES
time                 52.01 μs   (51.94 μs .. 52.08 μs)
                     1.000 R²   (1.000 R² .. 1.000 R²)
mean                 52.10 μs   (52.04 μs .. 52.18 μs)
std dev              248.7 ns   (219.4 ns .. 287.2 ns)

benchmarking KES/Sum6KES/updateKES
time                 18.14 μs   (18.06 μs .. 18.22 μs)
                     1.000 R²   (1.000 R² .. 1.000 R²)
mean                 18.12 μs   (18.07 μs .. 18.16 μs)
std dev              155.1 ns   (135.5 ns .. 179.5 ns)

benchmarking KES/CompactSum6KES/genKey
time                 1.205 ms   (1.204 ms .. 1.206 ms)
                     1.000 R²   (1.000 R² .. 1.000 R²)
mean                 1.206 ms   (1.205 ms .. 1.207 ms)
std dev              3.856 μs   (2.701 μs .. 5.840 μs)

benchmarking KES/CompactSum6KES/signKES
time                 18.99 μs   (18.92 μs .. 19.08 μs)
                     1.000 R²   (1.000 R² .. 1.000 R²)
mean                 18.96 μs   (18.94 μs .. 19.00 μs)
std dev              81.94 ns   (55.79 ns .. 129.6 ns)

benchmarking KES/CompactSum6KES/verifyKES
time                 52.43 μs   (52.32 μs .. 52.54 μs)
                     1.000 R²   (1.000 R² .. 1.000 R²)
mean                 52.42 μs   (52.38 μs .. 52.48 μs)
std dev              156.8 ns   (122.9 ns .. 198.0 ns)

benchmarking KES/CompactSum6KES/updateKES
time                 17.99 μs   (17.95 μs .. 18.03 μs)
                     1.000 R²   (1.000 R² .. 1.000 R²)
mean                 18.02 μs   (17.99 μs .. 18.06 μs)
std dev              111.2 ns   (95.31 ns .. 129.7 ns)

benchmarking VRF/SimpleVRF/genKey
time                 611.2 ns   (608.8 ns .. 614.2 ns)
                     1.000 R²   (0.999 R² .. 1.000 R²)
mean                 616.9 ns   (612.7 ns .. 621.6 ns)
std dev              14.72 ns   (11.66 ns .. 20.32 ns)
variance introduced by outliers: 31% (moderately inflated)

benchmarking VRF/SimpleVRF/eval
time                 28.49 ms   (28.41 ms .. 28.60 ms)
                     1.000 R²   (1.000 R² .. 1.000 R²)
mean                 28.51 ms   (28.43 ms .. 28.56 ms)
std dev              122.8 μs   (70.05 μs .. 205.8 μs)

benchmarking VRF/SimpleVRF/verify
time                 24.30 ms   (24.08 ms .. 24.49 ms)
                     1.000 R²   (1.000 R² .. 1.000 R²)
mean                 24.44 ms   (24.36 ms .. 24.51 ms)
std dev              173.2 μs   (119.0 μs .. 227.8 μs)

benchmarking VRF/PraosVRF/genKey
time                 18.40 μs   (18.38 μs .. 18.43 μs)
                     1.000 R²   (1.000 R² .. 1.000 R²)
mean                 18.38 μs   (18.35 μs .. 18.42 μs)
std dev              129.3 ns   (98.05 ns .. 194.1 ns)

benchmarking VRF/PraosVRF/eval
time                 166.0 μs   (165.7 μs .. 166.3 μs)
                     1.000 R²   (1.000 R² .. 1.000 R²)
mean                 166.7 μs   (166.4 μs .. 166.9 μs)
std dev              775.8 ns   (656.4 ns .. 924.5 ns)

benchmarking VRF/PraosVRF/verify
time                 208.5 μs   (208.3 μs .. 208.7 μs)
                     1.000 R²   (1.000 R² .. 1.000 R²)
mean                 208.4 μs   (208.3 μs .. 208.5 μs)
std dev              387.9 ns   (309.5 ns .. 512.6 ns)

Benchmark bench-crypto: FINISH
```

Summary of the performance-critical ones:
* DSIGN ed25519 verify: 49.5 μs
* KES verify: 52 μs
* VRF verify: 208.5 μs

File: cardano-crypto-tests/src/Bench/Crypto/DSIGN.hs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+             , Signable v a
+             , ExampleSignable v a
+             , NFData (SignKeyDSIGN v)
+             , NFData (VerKeyDSIGN v)
+             , NFData (SigDSIGN v)
+             )
+          => Proxy v
+          -> String
+          -> Benchmark
+bench_dsign _ lbl =
```
Comment (lehins) [inline cardano-crypto-tests/src/Bench/Crypto/DSIGN.hs:48]: Let's follow the naming scheme of the library and Haskell convention. I know other benchmarks already use this scheme, but I am suggesting we should fix it in all benchmarks
```suggestion
benchDSIGN :: forall v a
           . ( DSIGNAlgorithm v
             , ContextDSIGN v ~ ()
             , Signable v a
             , ExampleSignable v a
             , NFData (SignKeyDSIGN v)
             , NFData (VerKeyDSIGN v)
             , NFData (SigDSIGN v)
             )
          => Proxy v
          -> String
          -> Benchmark
benchDSIGN _ lbl =
```

File: cardano-crypto-tests/src/Bench/Crypto/DSIGN.hs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import Cardano.Crypto.DSIGN.Ed25519
+import Cardano.Crypto.DSIGN.EcdsaSecp256k1
+import Cardano.Crypto.DSIGN.SchnorrSecp256k1
+import Cardano.Crypto.Hash.Blake2b
+
+import Criterion
+
+import Bench.Crypto.BenchData
+
+{- HLINT ignore "Use camelCase" -}
```
Comment (lehins) [inline cardano-crypto-tests/src/Bench/Crypto/DSIGN.hs:26]:
```suggestion
```

File: cardano-crypto-tests/src/Bench/Crypto/DSIGN.hs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+
+benchmarks :: Benchmark
+benchmarks = bgroup "DSIGN"
+  [ bench_dsign (Proxy :: Proxy Ed25519DSIGN) "Ed25519"
+  , bench_dsign (Proxy :: Proxy EcdsaSecp256k1DSIGN) "EcdsaSecp256k1"
+  , bench_dsign (Proxy :: Proxy SchnorrSecp256k1DSIGN) "SchnorrSecp256k1"
```
Comment (lehins) [inline cardano-crypto-tests/src/Bench/Crypto/DSIGN.hs:33]:
```suggestion
  [ benchDSIGN (Proxy :: Proxy Ed25519DSIGN) "Ed25519"
  , benchDSIGN (Proxy :: Proxy EcdsaSecp256k1DSIGN) "EcdsaSecp256k1"
  , benchDSIGN (Proxy :: Proxy SchnorrSecp256k1DSIGN) "SchnorrSecp256k1"
```

**Review by lehins — APPROVED** [review]
Looks great. Just one minor suggestion, which if you'd rather not bother with I can take care of in a subsequent PR

**Comment by dcoutts** [comment]
I'm happy to change it to camel case. That's not a change I added, rather it's something I kept as is, but I'm happy to convert it.

**Pull request merged**